### PR TITLE
Atl cstring operator !=

### PR DESCRIPTION
**Pull request by learn-more**
## Purpose

Implement 'operator !='

### CODE REVIEW

File: sdk/lib/atl/cstringt.h [hunk 1]
```diff
@@ -533,6 +533,43 @@ class CStringT :
         return str2.GetLength() == 1 && str2[0] == ch1;
     }
 
+    friend bool operator!=(const CStringT& str1, const CStringT& str2) throw()
```
Comment (HBelusca) [inline sdk/lib/atl/cstringt.h:536]: Hmm... Why not just calling the corresponding `operator==` and NOT-ing the result?

Comment (learn-more) [reply sdk/lib/atl/cstringt.h:536]: because this was easier.

Comment (HBelusca) [reply sdk/lib/atl/cstringt.h:536]: Copy-pasting the code from operator== and manually changing every condition to its negative was "easier"??

Comment (learn-more) [reply sdk/lib/atl/cstringt.h:536]: yes.

File: sdk/lib/atl/cstringt.h [hunk 2]
```diff
@@ -533,6 +533,43 @@ class CStringT :
...
+
+    friend bool operator!=(PCYSTR psz1, const CStringT& str2) throw()
+    {
+        CStringT tmp(psz1, str2.GetManager());
+        return tmp.Compare(str2) != 0;
+    }
+
+    friend bool operator!=(XCHAR ch1, const CStringT& str2) throw()
+    {
+        return str2.GetLength() != 1 || str2[0] != ch1;
```
Comment (zefklop) [inline sdk/lib/atl/cstringt.h:570]: `return str2[0] != ch || str2[1} != '\0'`
feels more effcient.

Comment (learn-more) [reply sdk/lib/atl/cstringt.h:570]: It does, but some string implementations will assert when you access an item out of range, and I did want to make clear that there was not chance of that.

**Pull request merged**